realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -1553,6 +1553,10 @@ private void emitCreateOrUpdateUsingJsonObject(JavaWriter writer) throws IOExcep
         writer.emitEmptyLine();
     }
 
+    // FIXME: Since we need to check the PK in stream before create an object, this is now using copyToRealm instead of
```

Comment:
I am not sure we can ship this? It sounds like a major regression (at least for the ones using this API). 
I'm not sure what you mean by parsing the stream twice? Currently we convert to JSONObject, but that is the same as doing `copyToRealm`, just with JSON, no?